File: nestjs/src/business-rules/customers/handlers/update-customer.handler.ts
import {
  ICustomerRepository,
  ICustomerRepositoryName,
} from '@domain/repositories';
import { Inject, Injectable, NotFoundException } from '@nestjs/common';
import { IUpdateCustomerRequest } from '../requests';
import { ICustomerResponse } from '../responses';
import { IUpdateCustomerHandler } from './iupdate-customer.handler';

@Injectable()
export class UpdateCustomerHandler implements IUpdateCustomerHandler {
  constructor(
    @Inject(ICustomerRepositoryName) private repository: ICustomerRepository,
  ) {}

  async execute(request: IUpdateCustomerRequest): Promise<ICustomerResponse> {
    let customer = await this.repository.findById(request.id);
    if (!customer) {
      throw new NotFoundException('Customer not found');
    }

    customer = await this.repository.update({
      id: request.id,
      name: request.name,
      email: request.email,
      birthDate: request.birthDate,
    });

    const response: ICustomerResponse = {
      payload: {
        id: customer.id,
        name: customer.name,
        email: customer.email,
        birthDate: customer.birthDate,
      },
    };

    return response;
  }
}

File: dotnet/BusinessRules/Customers/Handlers/UpdateCustomerHandler.cs
using dotnet.BusinessRules.Customers.Requests;
using dotnet.BusinessRules.Customers.Responses;
using dotnet.BusinessRules.Exceptions;
using dotnet.Domain.Repositories;

namespace dotnet.BusinessRules.Customers.Handlers
{
    public class UpdateCustomerHandler : IUpdateCustomerHandler
    {
        private readonly ICustomerRepository _customerRepository;

        public UpdateCustomerHandler(
            ICustomerRepository customerRepository
        )
        {
            _customerRepository = customerRepository;
        }

        public CustomerResponse ExecuteAsync(UpdateCustomerRequest request)
        {
            var customer = _customerRepository.FindById(request.Id);
            if (customer == null)
            {
                throw new NotFoundException("Cliente não encontrado");
            }

            customer.Name = request.Name;
            customer.Email = request.Email;
            customer.BirthDate = request.BirthDate;

            customer = _customerRepository.Update(customer);

            return new CustomerResponse
            {
                Payload = new CustomerItemResponse
                {
                    Id = customer.Id,
                    Name = customer.Name,
                    Email = customer.Email,
                    BirthDate = customer.BirthDate
                }
            };
        }
    }
}